index.js

```diff
@@ -63,6 +65,63 @@ async function fetchAllCustomers(defaultHeaders) {
     return customers
 }
 
+async function capturePaidInvoices(defaultHeaders, customerIgnoreRegex) {
+    let invoices = []
+
+    let paginationParam = ''
+    let invoiceJson = { has_more: true }
+
+    const today = new Date()
+    const firstDayThisMonth = new Date(today.getFullYear(), today.getMonth(), 1)
+    const firstDayNextMonth = new Date(today.getFullYear(), today.getMonth() + 1, 1)
+
+    const dateParam = `created[gte]=${parseInt((firstDayThisMonth.getTime() / 1000).toFixed(0))}&created[lt]=${parseInt(
+        (firstDayNextMonth / 1000).toFixed(0)
+    )}`
```

Comment:
I wonder if we should also filter upcoming invoices by creation date lower than `firstDayNextMonth`

Reply:
we are w/ this param: `&created[lt]=${parseInt(
        (firstDayNextMonth / 1000).toFixed(0)`

Reply:
I meant upcoming invoices:

```
                const upcomingInvoiceResponse = await fetchWithRetry(
                    `https://api.stripe.com/v1/invoices/upcoming?customer=${customer.id}`,
                    global.defaultHeaders
                )
```

wasn't a comment on the code sorry, just a thought I had in general

Reply:
Ahhh got it. Yea with upcoming invoices, I'm less clear about the right filters. The decision right now is to just rely on stripe to give us what should be considered upcoming. We went back a few times last week when discussing reporting for the board slides.  With _our_ stripe account, it looks like upcoming invoices capture things we want to know about (e.g. we aren't capturing multiple future invoices or some invoice that's months out).

Given this I think the best thing for now is to 1) track payments received 2) track upcoming invoices and see how those numbers grow and change over the course of a month.

If it looks like these numbers look like they're getting out of whack we can take another pass

Reply:
I guess we have a date for them anyway so can always filter out next month's stuff in the UI